File: Programmers/src/짝지어제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        Stack<Character> stack = new Stack<Character>();
        
        for(int i=0; i<s.length(); i++){
            char ch = s.charAt(i);
            
            if(!stack.isEmpty() && ch == stack.peek())
                stack.pop();
            else
                stack.add(ch);
        }
        
        return stack.isEmpty() ? 1 : 0;
    }
}

File: Programmers/src/JS/짝지어제거하기.js
function solution(s)
{
    let answer = 0;
    let stack = [];
    
    for(let i=0; i<s.length; i++){
        if(stack[stack.length-1] === s.charAt(i))
            stack.pop();
        else
            stack.push(s.charAt(i));
    }
    
    answer = stack.length === 0 ? 1 : 0;
    return answer;
}